### PR TITLE
Api minor compat

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by tacaswell** [comment]
@arkilic did these changes get deployed?

**Comment by arkilic** [comment]
Nope, waiting for you to tag it. Is there anything I need to do for metadatastore library to insert events with the new schema or it is part of master already?

File: metadataservice/server/engine.py [hunk 1]
```diff
@@ -272,7 +272,7 @@ class EventHandler(DefaultHandler):
     def get(self):
         database = self.settings['db']
         query = utils.unpack_params(self)
-        docs = database['event'].find(query)
+        docs = database['event'].find(query, sort=[('time', 1)])
```
Comment (tacaswell) [inline metadataservice/server/engine.py]: This seems like an API change that needs to be fixed.

Comment (arkilic) [inline metadataservice/server/engine.py]: Sorting on cursor is a performance bottleneck, that's why I have left it out. We can sort locally on the client side

Comment (tacaswell) [inline metadataservice/server/engine.py]: ok

**Comment by arkilic** [comment]
@tacaswell Time to resurrect this?

**Comment by tacaswell** [comment]
I think the sort is a no-brainer, not  sure about ujson though.

File: metadataservice/server/engine.py (hunk 1 above)
Comment (arkilic) [inline metadataservice/server/engine.py]: --future me: only a bottleneck if cursor is holding a large number of documents

Comment (tacaswell) [inline metadataservice/server/engine.py]: can we index on time to get around that bottle neck?

Comment (arkilic) [inline metadataservice/server/engine.py]: We already index on time

**Comment by danielballan** [comment]
Right. What about @tacaswell's suggestion that we use metadataclient for retrieval (where losing a little precision via ujson is less of a worry) but metadatastore for insert -- for now, anyway?

**Comment by arkilic** [comment]
Or we could use simplejson or any other json package for inserts and ujson for retrieval. We need speed for the return, not for the inserts.

**Comment by danielballan** [comment]
I seem to remember slow inserts also being a problem for @klauer at one point. Could we have all of the above?

* metadataservice/client with fast, approximate ujson -- retrieval only
* metadataservice/client with slow, exact json package (is exactness even possible in a float -> string -> float roundtrip?) -- insertion or retrieval
* metadatastore -- insertion only

We can retire metadatastore when we are sure we have a sufficiently fast server/client replacement. The search for that replacement does not have to hold up the deployment of the whole.

**Comment by ericdill** [comment]
> I seem to remember slow inserts also being a problem for @klauer at one point.  

Wasn't that solved by the bulk event insertion in metadatastore?

**Comment by danielballan** [comment]
@ericdill Yes, using bulk insertion improved the performance. Now, adding a JSON serialization step might degrade the performance to an unacceptable extent.

**Comment by arkilic** [comment]
Also, @klauer and I think retrieving 100 million events in less than 10 seconds is the requirement in the long run (yay, big data). I have done some interesting work to accomplish this and initial results seem promising. I halted the work on this as I had to work on other more needed things but I will resume work later.

File: metadataservice/server/engine.py (hunk 1 above)
Comment (tacaswell) [inline metadataservice/server/engine.py]: We need the compound indexes to make this efficent.

**Comment by arkilic** [comment]
Outdated after [major rewrite ](https://github.com/NSLS-II/metadataservice/pull/5)

**Pull request closed without merging**